### PR TITLE
Improving SQL performance when retrieving reminders

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

Moved the logic from php to sql

Tested On: 
- Ubuntu 14.04 MySQL 5.5 and PHP 5.5
- Windows Server 2008 r2 with SQL Server 2008

## Motivation and Context

* Removes the need for multiple queries
* Improves performance

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/Reminders/Reminder.php [hunk 1]
```diff
@@ -292,84 +303,339 @@ public static function addNotifications(jsAlerts $alert, $checkDecline = true)
...
+         */
+        // Generate Related to (parent)queries
+        $callParentSelect = PHP_EOL;
+        $callParentJoin = PHP_EOL;
+        $meetingParentSelect = PHP_EOL;
+        $meetingParentJoin = PHP_EOL;
+
+
+        switch ($sugar_config['dbconfig']['db_type']) {
+            case "mssql":
```
Comment (mattlorimer) [inline modules/Reminders/Reminder.php]: limitQuery ?

Comment (daniel-samson) [inline modules/Reminders/Reminder.php]: I have update my query so that the db manager handles the db specific statements.

**Comment by sanchezfauste** [comment]
We have tested these modifications on several SuiteCRM installations and the performance improvement has been incredible. Before applying the improvement the response time in each action performed was about 30 seconds. Now the response time is less than one seccond.

So, the question is: Why is this not merged?

**Comment by daniel-samson** [comment]
@Dillon-Brown can you please assess this pull request?

**Comment by sanchezfauste** [comment]
We have detected that the remaining time to trigger the alert is not calculated correctly.

The problem is that variable `alertDateTimeNow` should be populated with current time in GMT format and now it's informed using current time in current_user datetime.

```diff
--- a/modules/Reminders/Reminder.php   2017-11-14 10:08:18.739607510 +0100
+++ b/modules/Reminders/Reminder.php      2017-11-14 10:09:48.468896601 +0100
@@ -278,9 +278,8 @@
         }
 
         // Create separate variable to hold timedate value
-        // These timedates need to be in the user time zone as the
-        // datetime returned by the Bean below is in the user time zone
-        $alertDateTimeNow = $timedate->getNow(true)->asDb(false);
+        // These timedates need to be in GMT
+        $alertDateTimeNow = $timedate->getNow()->asDb(false);
 
         // cn: get a boundary limiter
         $dateTimeMax = $timedate->getNow(true)->modify("+{$app_list_strings['reminder_max_time']} seconds")->asDb(false);
```
Apart from that, checking the code we have seen that variables `dateTimeNow` and `dateTimeMax` of `addNotifications` function are declared but not used. Maybe they could be safely deleted.

**Pull request closed without merging**